src/challenges/counting_duplicate.js

```diff
@@ -0,0 +1,20 @@
+function duplicateCount(text) {
+  text = text.toLowerCase()
+  let charCounter = {}
+  let total = 0
+  for (const x of text) {
+    if (x in charCounter) {
+      charCounter[x] += 1
+    } else {
+      charCounter[x] = 1
+    }
+  }
```

Comment:
Use text.split().reduce(() instead.

Reply:
i understand that text.split() will turn the text into an array but i don't understand how to use text.split().reduce()
i struggled a bit building the object to begin with.

Reply:
[reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) is an array method that takes 2 arguments, a function for the first and an initial value for the second.  The function needs to take 2 values, the first is the accumulator (or the thing you are constructing's initial value) and the second currentValue (a single item from the list)

```javaScript
let counts = text.split("").reduce(
  (charCounter, char) => 
    if (char in charCounter) {
      charCounter[char] += 1
    } else {
      charCounter[char] = 1
    }
    return charCounter
  },
  {}
)
```

See if the above works for you and then try to convert the one below too.  It isn't about this way being _better_, it's about developing a feel and understanding of all the builtin methods.

Reply:
why the {} at the end?

Reply:
> [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) is an array method that takes 2 arguments, a function for the first **and an initial value for the second.**

`{}` is the starting value for `charCounter`